homework/index.js

```diff
@@ -30,17 +36,91 @@
+    createAndAppend('h1', root, { text: error.message, class: `alert-error` });
+  }
+
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepos(repo, div) {
+    div.innerHTML = '';
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Name', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: `_blank`,
     });
+    addRow(tbody, 'Description', repo.description);
+    addRow(tbody, 'Forks', repo.forks);
+    addRow(tbody, 'Updated', new Date(repo.updated_at).toLocaleString('en-GB'));
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  function renderContributions(repo, ul) {
+    fetchJSON(repo.contributors_url)
+      .then(contributors => {
+        ul.innerHTML = '';
+        createAndAppend('h3', ul, { text: `Contributors:` });
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul);
+          const a = createAndAppend('a', li, { href: contributor.html_url, target: '_blank' });
+          const table = createAndAppend('table', a);
+          const tbody = createAndAppend('tbody', table);
+          const tr1 = createAndAppend('tr', tbody);
+          createAndAppend('img', tr1, { src: contributor.avatar_url });
+          createAndAppend('td', tr1, { text: contributor.login });
+          createAndAppend('td', tr1, { text: contributor.contributions });
+        });
+      })
+      .catch(error => renderError(error));
+  }
+
+  function createOptionElements(repositories, select) {
+    repositories
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repo, index) => {
+        createAndAppend('option', select, { text: repo.name, value: index });
+      });
+  }
+
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header);
+    const mainContainer = createAndAppend('div', root, { id: `main-container` });
+    const tableContainer = createAndAppend('div', mainContainer, { id: 'table-container' });
+    const listContainer = createAndAppend('div', mainContainer, { id: 'list-container' });
+    const ul = createAndAppend('ul', listContainer, { id: 'list-contributions' });
+
+    fetchJSON(url)
+      .then(repositories => {
+        createOptionElements(repositories, select);
+
+        function setDefault() {
+          renderRepos(repositories[0], tableContainer);
+          renderContributions(repositories[0], ul);
+        }
+        setDefault();
```

Comment:
See my comments on this useless function in the week 1 feedback.

Reply:
Oh I didn't see your last reply on homework week 1. Now I understand thank you